### PR TITLE
docker: add Dockerfile which runs starfish by default

### PR DESCRIPTION
**Pull request by joshmoore**
Happy to also enable building of this in .travis if useful.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Development dockerfile for starfish, primarily
+## focused on the running of tests but the executables
+## are installed to ~starfish/.local/bin
+##
+## Examples:
+##
+##   docker build -t starfish .
```
Comment (ambrosejcarr) [inline Dockerfile]: This is really great. How would you feel about adding a short explanation to each command? For those less familiar with docker, they might not know that these: (1) run help, (2) are equivalent to calling `starfish`, and (3) run verbose tests. 

Once this is built, what do you think about us pushing this up to quay or dockerhub and having it autobuild on new commits to master?

**Comment by ambrosejcarr** [comment]
I haven't used travis to build Dockerfiles before -- what would that accomplish?

File: Dockerfile (hunk 1 above)
Comment (joshmoore) [reply Dockerfile]: More than happy to add the explanations. One question I had myself was whether you wanted the tests run by default or `starfish`. As for Dockerhub, by all means. Let me know to what extent you want me to create the org and/or repository.

**Comment by joshmoore** [comment]
> I haven't used travis to build Dockerfiles before -- what would that accomplish?

If nothing else, it's testing the Dockerfile continues to build for all PRs. (Docker Hub will only tell you for well-defined branches). Additionally if interested, logic can be refactored out of `.travis.yml` in favor of `Dockerfile` so that users can make use of the steps locally.

File: Dockerfile (hunk 1 above)
Comment (ambrosejcarr) [reply Dockerfile]: I think `starfish` would be preferable. @ttung what do you think?

**Comment by joshmoore** [comment]
Suggestions pushed.

**Comment by ttung** [comment]
> Happy to also enable building of this in .travis if useful.

Does that mean .travis will build the docker image?  Or fire up docker and run things inside the docker instance?

**Comment by joshmoore** [comment]
@ttung : both are an option. If you have a Dockerfile in the repo, I'd suggest at least building the image so that you know it's still working for all PRs.

**Comment by joshmoore** [comment]
```
$ docker build -t spacetx/starfish .
...
Successfully tagged spacetx/starfish:latest
```

**Comment by ttung** [comment]
lgtm.  i hope this doesn't add much to the build time. :)

**Comment by joshmoore** [comment]
PR 285 was 6 minutes; this one 5. Apparently it's saving time... ;)

**Pull request merged**